Preserve suffix text when applying function suggestion

miqh opened the pull request:
Hello,

This pull request attempts to address #13885.

Exactly how a chosen suggestion applies is a somewhat contentious area depending on context.

Sometimes, its intuitive to replace the entire contents of the current context with the suggestion picked. A common case is typing keywords that are the subsequence contained inside a suggestion.

```
tr| -> pick "metric" -> metric|
me|tr -> pick "metric" -> metric|
```

However, for another type of suggestion like a "function", doing the same isn't particularly desirable.

```
to|metric -> pick "topk" -> topk(|) 
```

So, one workaround here is to make any "function"-like suggestion preserve any suffix text.

```
to|metric -> pick "topk" -> topk(|)metric
```

CC: @davkal

---

Also, I don't know if you've seen the [latest release (0.43.0)][1] of Slate, but they've now got a breaking change by removing the `Change` object. Probably a good time to think about refactoring the query field.

[1]: https://github.com/ianstormtaylor/slate/blob/24af6113dc3372fcd8779039b599f4175670d4ae/packages/slate/Changelog.md#0430--october-27-2018

davkal reviewed (changes requested):
While I like the fix, I think the behavior is still not right. Consider
```
to|metric -> pick "topk" -> topk(|)metric
```
Afterwards I need to fix the parens manually, which is a couple of keystrokes. If accepting the function does not complete the parens and I add them manually, they are correctly placed around the next term. This is more desirable IMO. In light of the alternate aggregation syntax (`sum by...`) we might as well not complete the parens when accepting functions for now.

miqh commented:
@davkal, yep, agreed—that UX would interrupt flow. :slightly_smiling_face:

What's your preference here for how to tackle omitting braces from function-like suggestions?

* Manually change every single `insertText` entry in `promql.ts`
* Have the `applyTypeahead` function strip them out when the need arises

davkal commented:
> Manually change every single insertText entry in promql.ts

I'd "manually" change it in `promql.ts`.

miqh committed:
Preserve suffix text when applying function suggestion

Updates all PromQL function-like suggestions to omit parentheses
when inserted. Such insertions also preserve any suffix text.

Related: #13885

davkal commented:
This works great now. I'm a bit on the fence if the function label text should now include the parens to indicate that it's part of the function group, e.g., `sum()`, while only applying `sum`. But perhaps that's confusing.

Thank you @miqh 🎉
